src/pipdeptree/_models/package.py

```diff
@@ -3,33 +3,36 @@
 import re
 from abc import ABC, abstractmethod
 from importlib import import_module
-from importlib.metadata import PackageNotFoundError, metadata, version
+from importlib.metadata import Distribution, PackageNotFoundError, metadata, version
 from inspect import ismodule
-from typing import TYPE_CHECKING
+from pathlib import Path
+from typing import TYPE_CHECKING, Any
 
-from pip._vendor.pkg_resources import Requirement  # noqa: PLC2701
+from packaging.requirements import Requirement
+from packaging.specifiers import SpecifierSet
+from pip._internal.models.direct_url import DirectUrl  # noqa: PLC2701
+from pip._internal.utils.egg_link import egg_link_path_from_sys_path  # noqa: PLC2701
 
 if TYPE_CHECKING:
-    from pip._internal.metadata import BaseDistribution
-    from pip._vendor.pkg_resources import DistInfoDistribution
+    from importlib.metadata import Distribution
 
 
 def pep503_normalize(name: str) -> str:
-    return re.sub("[-_.]+", "-", name)
+    return re.sub("[-_.]+", "-", name).lower()
+
+
+def contains_extra(marker: str) -> bool:
+    return re.search(r"\bextra\s*==", marker)
 
 
 class Package(ABC):
     """Abstract class for wrappers around objects that pip returns."""
 
     UNKNOWN_LICENSE_STR = "(Unknown license)"
 
-    def __init__(self, obj: DistInfoDistribution) -> None:
-        self._obj: DistInfoDistribution = obj
-        self.key = pep503_normalize(obj.key)
-
-    @property
-    def project_name(self) -> str:
-        return self._obj.project_name  # type: ignore[no-any-return]
+    def __init__(self, _project_name: str) -> None:
+        self.project_name = _project_name
+        self.key = pep503_normalize(_project_name)
```

Comment:
```suggestion
    def __init__(self, project_name: str) -> None:
        self.project_name = project_name
        self.key = pep503_normalize(project_name)
```

Reply:
Yes, but Python doesn't treat `project_name` and `self.project_name` as the same thing and since it's only used in this small constructor it is fine.

Reply:
Ok, done this.